### PR TITLE
Fuzz: Check individual and package transaction invariants

### PR DESCRIPTION
**Pull request by instagibbs**
Poached from https://github.com/bitcoin/bitcoin/pull/26711 since that PR is being split apart, and modified to match current behavior.

### CODE REVIEW

File: src/test/util/txmempool.h [hunk 1]
```diff
@@ -36,4 +38,11 @@ struct TestMemPoolEntryHelper {
     TestMemPoolEntryHelper& SigOpsCost(unsigned int _sigopsCost) { sigOpCost = _sigopsCost; return *this; }
 };
 
+/** Check expected properties for every PackageMempoolAcceptResult, regardless of value. Returns
+ * false if an error occurs, true otherwise.  String s will be populated with an error message if
+ * the return value is false.  If mempool is provided, checks that the expected transactions are in
+ * mempool (this should be set to nullptr for a test_accept).
+*/
+bool CheckPackageMempoolAcceptResult(const Package& txns, const PackageMempoolAcceptResult& result, bool expect_valid,
+                                            const CTxMemPool* mempool, std::string& s);
```
Comment (dergoegge) [inline src/test/util/txmempool.h:47]:
```suggestion
std::optional<std::string> CheckPackageMempoolAcceptResult(const Package& txns, const PackageMempoolAcceptResult& result, bool expect_valid,
                                            const CTxMemPool* mempool);
```

Comment (instagibbs) [reply src/test/util/txmempool.h]: taken, thanks

File: src/test/util/txmempool.cpp [hunk 2]
```diff
@@ -36,3 +37,78 @@ CTxMemPoolEntry TestMemPoolEntryHelper::FromTx(const CTransactionRef& tx) const
 {
     return CTxMemPoolEntry{tx, nFee, TicksSinceEpoch<std::chrono::seconds>(time), nHeight, m_sequence, spendsCoinbase, sigOpCost, lp};
 }
+
+std::optional<std::string> CheckPackageMempoolAcceptResult(const Package& txns, const PackageMempoolAcceptResult& result, bool expect_valid,
+                                     const CTxMemPool* mempool)
```
Comment (glozow) [inline src/test/util/txmempool.cpp:42]: c1319395f74c0016b367e860050c0ff5a99dea85 nit: the whitespace here is weird, should probably align with parentheses

**Review by glozow — COMMENTED** [review]
ACK b847d48f70b5a67263c362f9d28ee6d092068e27, thanks for shaving!

File: src/test/util/txmempool.cpp (hunk 2 above)
Comment (instagibbs) [reply src/test/util/txmempool.cpp]: realigned

**Comment by glozow** [comment]
reACK fcb3069fa307942cf7f3edabcda1be96d615c91f, only whitespace changes

**Review by dergoegge — APPROVED** [review]
ACK fcb3069fa307942cf7f3edabcda1be96d615c91f

File: src/test/util/txmempool.cpp [hunk 3]
```diff
@@ -36,3 +37,80 @@ CTxMemPoolEntry TestMemPoolEntryHelper::FromTx(const CTransactionRef& tx) const
...
+                                                           bool expect_valid,
+                                                           const CTxMemPool* mempool)
+{
+    if (expect_valid) {
+        if (result.m_state.IsInvalid()) {
+            return strprintf("Package validation unexpectedly failed: %s", result.m_state.ToString());
+        }
+    } else {
+        if (result.m_state.IsValid()) {
+            strprintf("Package validation unexpectedly succeeded. %s", result.m_state.ToString());
```
Comment (instagibbs) [inline src/test/util/txmempool.cpp:52]: missing some `return` statements in this function

**Pull request merged**